packages/field-plugin/src/createFieldPlugin/createPluginActions/callbackQueue.ts

```diff
@@ -19,28 +19,19 @@ type CallbackMap = {
 }
 type CallbackType = keyof CallbackMap
 
-export const callbackQueue = () => {
+export const callbackQueue = (maxQueueLength: number) => {
   // eslint-disable-next-line functional/no-let
   let callbackMap: CallbackMap = {
     asset: [],
     context: [],
     stateChanged: [],
     loaded: [],
   }
-  // eslint-disable-next-line functional/no-let
-  let uuidIndex = 0
-  // TODO randomly generate instead
-  const uuid = (): CallbackId => {
-    uuidIndex = uuidIndex + 1
-    return uuidIndex
-  }
-
   const pushCallback = <T extends CallbackType>(
```

Comment:
I think it doesn't hurt to mutate `callbackMap` here. I'd just `callbackMap[type].push(..)` instead of restructuring the whole map. If we let ourselves mutate `callbackMap`, then it's easy to implement popping: either `array.pop()` or `array.shift()`